File: student/2.7.c
#include <stdio.h>
#include <string.h>

#define MAX_LEN 1000 // 定義最大字串長度

int main() {
    char lyrics[MAX_LEN] = "I'm at a payphone trying to call home\nAll of my change I spent on you\nWhere have the times gone baby\nIt's all wrong, where are the plans we made for two?"; // 歌詞
    char *token, *str_ptr;
    char delim[] = " ,.\n";
    int word_count = 0;
    int index = 0;
    char words[MAX_LEN][MAX_LEN] = {0}; // 二維陣列，存儲歌詞中的每個單詞
    int word_index[MAX_LEN][MAX_LEN] = {0}; // 二維陣列，存儲每個單詞在歌詞中的位置
    int word_freq[MAX_LEN] = {0}; // 陣列，存儲每個單詞出現的次數
    
    // 將歌詞中的單詞存入 words 陣列，同時記錄每個單詞在歌詞中的位置
    str_ptr = lyrics;
    while ((token = strtok(str_ptr, delim))) {
        strcpy(words[word_count], token);
        word_index[word_count][word_freq[word_count]] = index;
        word_count++;
        word_freq[word_count - 1]++;
        index += strlen(token) + 1;
        str_ptr = NULL;
    }
    
    // 輸出每個單詞出現的次數
    for (int i = 0; i < word_count; i++) {
        printf("%s: %d\n", words[i], word_freq[i]);
    }
    
    return 0;
}

File: student/2.7.py
music = "I'm at a payphone trying to call home\
All of my change I spent on you\
Where have the times gone\
Baby, it's all wrong, where are the plans we made for two?\
Yeah, I, I know it's hard to remember\
The people we used to be\
It's even harder to picture\
That you're not here next to me\
You say it's too late to make it\
But is it too late to try?\
And in our time that you wasted\
All of our bridges burned down\
I've wasted my nights\
You turned out the lights\
Now I'm paralyzed\
Still stuck in that time\
When we called it love\
But even the sun sets in paradise\
I'm at a payphone trying to call home\
All of my change I spent on you\
Where have the times gone\
Baby, it's all wrong, where are the plans we made for two?\
If happy ever after did exist\
I would still be holding you like this\
All those fairy tales are full of shit\
One more fucking love song, I'll be sick"

# 用空白分隔單字，將大寫轉為小寫，並刪除標點符號
words = music.lower().replace(',', '').replace('.', '').replace('?', '').replace('!', '').split()

word_dict = {}
for index, word in enumerate(words):
    if word not in word_dict:
        word_dict[word] = [index]
    else:
        word_dict[word].append(index)
# 計算每個單字出現次數
word_count = {}
for word in word_dict:
    word_count[word] = len(word_dict[word])

print(word_count)